src/Appwrite/Platform/Workers/Deletes.php

```diff
@@ -565,10 +572,17 @@ private function deleteProject(Database $dbForConsole, callable $getProjectDB, D
         ], $dbForConsole);
 
         // Delete metadata table
-        if (!\in_array($dsn->getHost(), $sharedTables)) {
+        if ($projectTables) {
             $dbForProject->deleteCollection(Database::METADATA);
-        } else {
+        } elseif ($sharedTablesV1) {
             $this->deleteByGroup(Database::METADATA, [], $dbForProject);
+        } elseif ($sharedTablesV2) {
+            $queries = \array_map(
+                fn ($id) => Query::notEqual('$id', $id),
+                $projectCollectionIds
+            );
+
+            $this->deleteByGroup(Database::METADATA, $queries, $dbForProject);
```

Comment:
Because all users share a single metadata document in V2, we need to make sure only project-specific tables are removed here